cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -562,7 +562,12 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                         // This was validated during normalization.
                         predicateValue = negatedPredicate.get(operator);
                         placeholder = "?";
-                        sqlParameters.add(String.valueOf(predicateValue));
+                        if (predicateValue instanceof Boolean){
```

Comment:
My point was we don't call `String.valueOf()` on the `Integer` types at the moment, but we would be after the change, which I don't think matters because I think they end up in the SQL statement as strings anyway, but I wasn't 100% sure if they actually needed to be `Integer` in the parameter list.